leshan-core/src/main/java/org/eclipse/leshan/senml/SenMLPack.java

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *     Boya Zhang - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.leshan.senml;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * The class representing the SenML Pack of LwM2M
+ */
+public class SenMLPack {
+
+    private String baseName = null;
```

Comment:
As I understand the specification `base fields` should be SenMLRecord attributes.
https://tools.ietf.org/html/rfc8428#section-2